examples/get_iss_data.py

```diff
@@ -16,97 +20,132 @@ def download(input_dir, url):
     z.extractall(input_dir)
 
 
-def load_tiff_stack(path):
-    stack = imread(path)
-    return stack
-
-
-def load_hyb_chan(input_dir, hyb, chan):
-    res = load_tiff_stack(input_dir + '{}/c{}.TIF'.format(hyb, chan))
-    return res
-
-
-def load_and_write_data(input_dir, output_dir):
-    prefix = 'fov_0'
-
-    hyb1 = [load_hyb_chan(input_dir, 1, c) for c in [2, 3, 4, 5]]
-    hyb2 = [load_hyb_chan(input_dir, 2, c) for c in [2, 3, 4, 5]]
-    hyb3 = [load_hyb_chan(input_dir, 3, c) for c in [2, 3, 4, 5]]
-    hyb4 = [load_hyb_chan(input_dir, 4, c) for c in [2, 3, 4, 5]]
-
-    dapi = load_tiff_stack(input_dir + 'DO/c1.TIF')
-    dots = load_tiff_stack(input_dir + 'DO/c2.TIF')
-
-    hyb_list = [hyb1, hyb2, hyb3, hyb4]
-
-    res = dict()
-    res['data'] = []
-    res['aux'] = []
-    res['metadata'] = dict()
-
-    for h, hyb in enumerate(hyb_list):
-        for c, img in enumerate(hyb):
-            fname = '{}_H_{}_C_{}.tiff'.format(prefix, h, c)
-            path = output_dir + fname
-            d = {'hyb': h,
-                 'ch': c,
-                 'file': fname}
-            res['data'].append(d)
-            imsave(path, img)
-
-    dapi_fname = '{}_{}.tiff'.format(prefix, 'dapi')
-    imsave(output_dir + dapi_fname, dapi)
-    res['aux'].append({'type': 'dapi', 'file': dapi_fname, 'format': "TIFF"})
-
-    dots_fname = '{}_{}.tiff'.format(prefix, 'dots')
+def build_hybridization_stack(input_dir):
+    default_shape = imread(os.path.join(input_dir, str(1), "c{}.TIF".format(1))).shape
+
+    imageslice = ImagePartition(["x", "y", "hyb", "ch"], {'hyb': 4, 'ch': 4}, default_shape, ImageFormat.TIFF)
+
+    for hyb in range(4):
+        for ch in range(4):
+            tile = Tile(
+                {
+                    'x': (0.0, 0.0001),
+                    'y': (0.0, 0.0001),
+                },
+                {
+                    'hyb': hyb,
+                    'ch': ch,
+                },
+            )
+            path = os.path.join(input_dir, str(hyb + 1), "c{}.TIF".format(ch + 2))
+            tile.set_source_fh_callable(
+                lambda _path=path: open(_path, "rb"),
+                ImageFormat.TIFF,
+            )
+            imageslice.add_tile(tile)
+
+    return imageslice
+
+
+def build_fov(input_dir, hybridization_stack_name, output_dir):
+    prefix = "fov_0"
+
+    nuclei = imread(input_dir + "DO/c1.TIF")
+    dots = imread(input_dir + "DO/c2.TIF")
+
+    experiment = {
+        'version': "0.0.0",
+        'hybridization_images': hybridization_stack_name,
+        'auxiliary_images': {},
+    }
+
+    nuclei_fname = "{}_{}.tiff".format(prefix, "nuclei")
+    imsave(output_dir + nuclei_fname, nuclei)
```

Comment:
could we `os.rename` instead of rewriting the files?

Reply:
Yes, but same as above.